File: src/Moryx.Maintenance.Web/src/databases/api/requests/RestoreDatabaseRequest.ts
/*
 * Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

import DatabaseConfigModel from "../../models/DatabaseConfigModel";

export default class RestoreDatabaseRequest {
    Config: DatabaseConfigModel;
    BackupFileName: string;
}

File: src/Moryx.Runtime.Endpoints/Databases/Endpoint/Request/RestoreDatabaseRequest.cs
// Copyright (c) 2023, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

using Moryx.Runtime.Endpoints.Databases.Endpoint.Models;

namespace Moryx.Runtime.Endpoints.Databases.Endpoint.Request
{
    /// <summary>
    /// Request to restore a database
    /// </summary>
    public class RestoreDatabaseRequest
    {
        /// <summary>
        /// Configuration of the database
        /// </summary>
        public DatabaseConfigModel Config { get; set; }

        /// <summary>
        /// Name of the backup that shall be restored
        /// </summary>
        public string BackupFileName { get; set; }
    }
}
